test/test-network/systemd-networkd-tests.py

```diff
@@ -173,15 +175,22 @@ def remove_log_file(self):
         if os.path.exists(dnsmasq_log_file):
             os.remove(dnsmasq_log_file)
 
-    def start_networkd(self, remove_state_files=True):
+    def start_networkd(self, sleep_sec=5, remove_state_files=True):
         if (remove_state_files and
             os.path.exists(os.path.join(networkd_runtime_directory, 'state'))):
             subprocess.check_call('systemctl stop systemd-networkd', shell=True)
             os.remove(os.path.join(networkd_runtime_directory, 'state'))
             subprocess.check_call('systemctl start systemd-networkd', shell=True)
         else:
             subprocess.check_call('systemctl restart systemd-networkd', shell=True)
-        time.sleep(5)
+        if sleep_sec > 0:
+            time.sleep(sleep_sec)
+
+    def wait_online(self, links_with_operstate, timeout='20s'):
+        str = wait_online_bin + ' --timeout ' + timeout
+        for link in links_with_operstate:
+            str += ' -i ' + link
+        subprocess.check_call(str, shell=True)
```

Comment:
There's no need to craft the string like that, or use the shell for that matter. Why not just:
```
args = [wait_online_bin, f' --timeout={timeout}] + \
           [f'--interface={link}' for for link in links_with_operstate]
subprocess.check_call(args)
```